src/lexer/TokenizerEngine.ts

```diff
@@ -52,13 +53,32 @@ export default class TokenizerEngine {
 
   private skipWhitespace(): void {
     WHITESPACE_REGEX.lastIndex = this.index;
+
     const matches = WHITESPACE_REGEX.exec(this.input);
     if (matches) {
+      this.updateLineCol(matches[0]);
+
       // Advance current position by matched whitespace length
       this.index += matches[0].length;
     }
   }
 
+  private updateLineCol(token: string) {
+    // if whitespace contains linebreaks
+    if (LINEBREAK_REGEX.test(token)) {
+      this.line++;
+      // increment line for each newline match
+      let lastIndex = 1;
+      while (LINEBREAK_REGEX.exec(token) !== null) {
+        this.line++;
+        lastIndex = LINEBREAK_REGEX.lastIndex;
+      }
+      this.col = token.length - lastIndex + 1;
```

Comment:
the + 1 is actually for the 1-based col index, not for the line match length
setting the `lastIndex` to 1 is to account for the line char but you're right that it doesn't account for `\r\n`

Reply:
the initial line increment is because `REGEX.test` and `REGEX.exec` have an interaction where the test counts as the first exec - ie. if there is only 1 line break, the exec loop would never start